fml/platform/posix/file_posix.cc

```diff
@@ -206,15 +206,12 @@ bool WriteAtomically(const fml::UniqueFD& base_directory,
     return false;
   }
 
-  FileMapping mapping(temp_file, {FileMapping::Protection::kWrite});
-  if (mapping.GetMutableMapping() == nullptr ||
-      data.GetSize() != mapping.GetSize()) {
+  ssize_t size = ::write(temp_file.get(), data.GetMapping(), data.GetSize());
```

Comment:
`size` may be less than the `data.GetSize()`. if its not -1 and less than `data.GetSize()`, there must be a loop that writes the entire data contents. Otherwise, the file contents are truncated and corrupt. Refer to the man page for `write` for details.

Reply:
I added the loop but I'm not clear on if we can assume the file is seekable as it relies on the file offset being updated:

```For a seekable file (i.e., one to which lseek(2) may be applied, for example, a regular file) writing takes place at the current file offset, and the file offset is incremented by the number of bytes actually written.```